### PR TITLE
Fix: native currency symbol on signature popup

### LINKED ISSUE #11950: It always sign with ETH symbol, but my actual symbol is other coin.
It always sign with ETH symbol, but my actual symbol is other coin.

image

### PR DESCRIPTION
**Pull request by MonarthS**
Fixes: #11950 

Explanation:  
- use native currency symbol instead of static ETH on signature popup

### CODE REVIEW

File: ui/components/app/signature-request-original/signature-request-original.component.js [hunk 1]
```diff
@@ -105,12 +106,13 @@ export default class SignatureRequestOriginal extends Component {
...
+      nativeCurrency,
       fromAccount: { balance },
-    } = this.state;
+    } = this.props;
 
-    const balanceInEther = conversionUtil(balance, {
+    const nativeCurrencyBalance = conversionUtil(balance, {
       fromNumericBase: 'hex',
       toNumericBase: 'dec',
       fromDenomination: 'WEI',
```
Comment (jpuri) [inline ui/components/app/signature-request-original/signature-request-original.component.js:118]: Is it possible for currency to have denomination different from `WEI` ?

Comment (MonarthS) [reply ui/components/app/signature-request-original/signature-request-original.component.js:118]: No.

File: ui/components/app/signature-request-original/signature-request-original.component.js [hunk 2]
```diff
@@ -105,12 +106,13 @@ export default class SignatureRequestOriginal extends Component {
   };
 
   renderBalance = () => {
-    const { conversionRate } = this.props;
     const {
+      conversionRate,
+      nativeCurrency,
       fromAccount: { balance },
-    } = this.state;
+    } = this.props;
```
Comment (danjm) [inline ui/components/app/signature-request-original/signature-request-original.component.js:113]: I think we should still get `fromAccount` from `this.state`

The component state is set on construction, but then not updated. And before this PR, the `fromAccount` was always just retrieved from state and not props. We should investigate why that was done that way and whether it requires correction, but that can be part of a separate PR.

**Review by danjm — CHANGES_REQUESTED** [review]
Thanks for the contribution. I have requested one change here.

File: ui/components/app/signature-request-original/signature-request-original.component.js (hunk 2 above)
Comment (MonarthS) [reply ui/components/app/signature-request-original/signature-request-original.component.js:113]: Actually I don't know, but I was checking other files [here](https://github.com/MetaMask/metamask-extension/blob/6187ab9b01ea757366a42d22b7292413c0f2b119/ui/pages/confirm-encryption-public-key/confirm-encryption-public-key.component.js#L112) also `props` is used instead of `state`.

Comment (MonarthS) [reply ui/components/app/signature-request-original/signature-request-original.component.js:113]: We can get fromAccount from `state`, but can't `nativeCurrency`.

**Review by jj-crypto — DISMISSED** [review]

**Comment by MonarthS** [comment]
@danjm any update on this?

**Comment by darkwing** [comment]
Hello @MonarthS!  Could you please rebase?  We'd like to have a good look at this one :)

**Comment by MonarthS** [comment]
@danjm , @darkwing you only accept PR from your orgaization? I opened thi PR almost 2 months ago & now someone opened same PR(#12696) today & it got merged.

**Comment by mcmire** [comment]
Hi @MonarthS, I'm going to close this since as you pointed out a PR we've merged already addresses this issue. I understand your consternation and I'm so sorry that this didn't get picked up sooner. We appreciate your submission however and hope that you continue to submit more fixes in the future.
